tiledb/sm/serialization/query.cc

```diff
@@ -1680,17 +1680,26 @@ Status query_from_capnp(
             }
           }
 
+          // attr_copy_state==nulptr models the case of deserialization code
+          // being called via the C API tiledb_query_deserialize (e.g. unit
+          // test serialization wrappers).
+          // When a rest_client exists, there is always a copy_state
           if (attr_copy_state == nullptr) {
             // Set the size directly on the query (so user can introspect on
             // result size).
-            if (existing_offset_buffer_size_ptr != nullptr)
-              *existing_offset_buffer_size_ptr =
-                  curr_offset_size + fixedlen_size_to_copy;
-            if (existing_buffer_size_ptr != nullptr)
-              *existing_buffer_size_ptr = curr_data_size + varlen_size;
-            if (nullable && existing_validity_buffer_size_ptr != nullptr)
-              *existing_validity_buffer_size_ptr =
+            // Subsequent incomplete submits will use the original buffer
+            // sizes members from the query in the beginning of the loop
+            // to calculate if user buffers have enough space to hold the data,
+            // here we only care that after data received from the wire is
+            // copied within the user buffers, the buffer sizes are accurate so
+            // user can introspect.
+            *query_buffer.buffer_size_ =
+                curr_offset_size + fixedlen_size_to_copy;
+            *query_buffer.buffer_var_size_ = curr_data_size + varlen_size;
```

Comment:
Why don't we update the var buffer size here?

Reply:
Oh I see I was looking at the one below, which doesn't because this massive if (var_size) { statement makes it challenging to review. Sorry.